rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Node.scala

```diff
@@ -102,27 +102,46 @@ sealed trait NodeDiff
 /**
  * Denote a change on the heartbeat frequency.
  */
-final case class ModifyNodeHeartbeatDiff(
-    id          : NodeId
-  , modHeartbeat: Option[SimpleDiff[Option[HeartbeatConfiguration]]]
-) extends NodeDiff with HashcodeCaching
+object ModifyNodeHeartbeatDiff{
+  def apply(id: NodeId,  modHeartbeat: Option[SimpleDiff[Option[HeartbeatConfiguration]]]) = ModifyNodeDiff(id,modHeartbeat, None, None, None)
+}
 
 /**
  * Diff on a change on agent run period
  */
-final case class ModifyNodeAgentRunDiff(
-    id         : NodeId
-  , modAgentRun: Option[SimpleDiff[Option[AgentRunInterval]]]
-) extends NodeDiff with HashcodeCaching
+object ModifyNodeAgentRunDiff{
```

Comment:
Those object are only used as syntax sugar when we are reading the old event (Could be replaced by ModifyNode(agentRun,None,None,None)